clients/java/src/main/java/io/camunda/zeebe/client/api/search/FlowNodeInstanceState.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.search;
+
+import io.camunda.zeebe.client.protocol.rest.FlowNodeInstanceFilterRequest;
+import io.camunda.zeebe.client.protocol.rest.FlowNodeInstanceItem;
+
+public enum FlowNodeInstanceState {
+  ACTIVE,
+  COMPLETED,
+  TERMINATED;
+
+  public static FlowNodeInstanceFilterRequest.StateEnum toProtocolState(
+      FlowNodeInstanceState value) {
+    return (value == null) ? null : FlowNodeInstanceFilterRequest.StateEnum.fromValue(value.name());
+  }
+
+  public static FlowNodeInstanceState fromProtocolState(FlowNodeInstanceItem.StateEnum value) {
+    return (value == null) ? null : FlowNodeInstanceState.valueOf(value.name());
```

Comment:
If the client is using an older version than the C8 cluster, the value can default to `UNKNOWN_DEFAULT_OPEN_API ` when the API returns an enum value that is not recognized by the protocol enum used by the client. In this scenario, should we map it to a predefined `UNKNOWN` enum value, or should we throw an explicit exception to indicate that this value is not supported? WDYT @mihail-ca @tmetzke?

Reply:
Currently [an exception is thrown](https://github.com/camunda/camunda/pull/25970/files#diff-ea8dac8514d4a25ce4625e646b4908230db960edb40537a66cfb3a4232761ec6R148-R155), when the protocol value is UNKNOWN_DEFAULT_OPEN_API. But honestly I don't know how this UNKNOWN_DEFAULT_OPEN_API enum is supposed to work, and how we should handle it.

Reply:
I generally like the idea of keeping the client forward-compatible by being more resilient and handling unknown values. However, if I am not mistaken, other parts of the client are not consistent with this either, e.g. the [PartitionInfo](https://github.com/camunda/camunda/blob/7e9b19bcd1abae8262706616b237796f241b3537/clients/java/src/main/java/io/camunda/zeebe/client/impl/response/PartitionInfoImpl.java#L30) you mentioned earlier in this PR.

If we want this, I'd propose to create a follow-up issue to make the client more resilient and forward-compatible because it should be consistent and there might be even more places where this is needed.

What do you think, @houssain-barouni?

Reply:
I guess we could add test cases that prove that the client throws an exception when the server actually returns an enum value we cannot handle, for the partition info and the flow node type and state enums.

Reply:
@tmetzke I added tests with exception [here](https://github.com/camunda/camunda/pull/25970/files#diff-ea8dac8514d4a25ce4625e646b4908230db960edb40537a66cfb3a4232761ec6R148). Or do you have something else in mind ? Thank you.

Reply:
That is quite "internal" as a test and it's valid. However, I was thinking more of a test case that fires a search request from the client, mocks the server response to an unknown enum, and asserts that the client throws an exception in that case. Using the OpenAPI-specific enum type is an implementation detail that might change with a new version of the OpenAPI generator, actually.

Reply:
I decided to follow @houssain-barouni's suggestion here and handle such cases gracefully. I also adjusted a bit how we do this in the existing places by introducing a dedicated enum value for such cases to allow differentiating those cases. I also added debug logs for such cases in the client.